bears/vcs/CommitBear.py

```diff
@@ -0,0 +1,392 @@
+        """
+        words = nltk.word_tokenize(nltk.sent_tokenize(paragraph)[0])
+        # VBZ : Verb, 3rd person singular present, like 'adds', 'writes'
+        #       etc
+        # VBD : Verb, Past tense , like 'added', 'wrote' etc
+        # VBG : Verb, Present participle, like 'adding', 'writing'
+        word, tag = nltk.pos_tag(['I'] + words)[1:2][0]
+        if(tag.startswith('VBZ') or
+           tag.startswith('VBD') or
+           tag.startswith('VBG') or
+           word.endswith('ing')):  # Handle special case for VBG
+            return (word, tag)
+        else:
+            return None
+
+    def check_body(self, body,
+                   body_line_length: int = 72,
+                   force_body: bool = False,
+                   ignore_length_regex: typed_list(str) = (),
+                   body_regex: str = None
+                   ):
+        """
+        Checks the given commit body.
+
+        :param body:                The body of the commit message of HEAD.
+        :param body_line_length:    The maximum line-length of the body. The
+                                    newline character at each line end does not
+                                    count to the length.
+        :param force_body:          Whether a body shall exist or not.
+        :param ignore_length_regex: Lines matching each of the regular
+                                    expressions in this list will be ignored.
+        :param body_regex:          If provided, checks the presence of regex
+                                    in the commit body.
+        """
+        if len(body) == 0:
+            if force_body:
+                yield Result(self, 'No commit message body at HEAD.')
+            return
+
+        if body[0] != '\n':
+            yield Result(self, 'No newline found between shortlog and body at '
+                               'HEAD commit. Please add one.')
+            return
+
+        if body_regex and not re.fullmatch(body_regex, body.strip()):
+            yield Result(self, 'No match found in commit message for the '
+                               'regular expression provided: %s' % body_regex)
+
+        body = body.splitlines()
+        ignore_regexes = [re.compile(regex) for regex in ignore_length_regex]
+        if any((len(line) > body_line_length and
+                not any(regex.search(line) for regex in ignore_regexes))
+               for line in body[1:]):
+            yield Result(self, 'Body of HEAD commit contains too long lines. '
+                               'Commit body lines should not exceed {} '
+                               'characters.'.format(body_line_length))
+
+    def check_issue_reference(self, body,
+                              body_close_issue: bool = False,
+                              body_close_issue_full_url: bool = False,
+                              body_close_issue_on_last_line: bool = False,
+                              body_enforce_issue_reference: bool = False
```

Comment:
you need to learn `git diff -u -C `

Reply:
This is *awesome*

Thanks a lot :smile: